src/main/java/seedu/duke/utils/Parser.java

```diff
@@ -469,31 +470,45 @@ private boolean singleDelimiterChecker(String stringToCheck) {
      * @param matcher matcher object to match the entire input sequence against the regex pattern.
      * @return an array of the found matches.
      **/
-    private String[] splitArgsForEditCommand(Matcher matcher) {
+    private String[] splitArgsForEditCommand(Matcher matcher) throws NumberFormatException {
 
         String indexOfItem = null;
         String itemDescription = null;
         String itemPrice = null;
         String itemQuantity = null;
+        Double testprice = 0.0; //used to check price in acceptable format
+        int testQuantity = 0; //used to check if quantity is in acceptable format
+
 
         while (matcher.find()) {
 
             if (matcher.group("index") != null) {
-                indexOfItem = matcher.group("index");
+                indexOfItem = matcher.group("index").trim();
             }
 
             if (matcher.group("description") != null) {
-                itemDescription = matcher.group("description");
+                itemDescription = matcher.group("description").trim();
             }
 
             if (matcher.group("price") != null) {
-                itemPrice = matcher.group("price");
+                if (matcher.group(0).contains("-")) {
+                    throw new NumberFormatException();
```

Comment:
Perhaps you can just have an if statement to check whether the result of the parseDouble(itemPrice) is lesser than 0, if yes then return new IncorrectCommand?

Reply:
okiee!! :D